hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/client/rpc/TestBlockDataStreamOutput.java

```diff
@@ -41,154 +45,185 @@
+    MiniOzoneCluster cluster = MiniOzoneCluster.newBuilder(conf)
         .setNumDatanodes(5)
         .setDatanodeFactory(UniformDatanodesFactory.newBuilder()
             .setCurrentVersion(DN_OLD_VERSION)
             .build())
         .build();
+    cluster.waitForPipelineTobeReady(HddsProtos.ReplicationFactor.THREE,
+        180000);
     cluster.waitForClusterToBeReady();
-    //the easiest way to create an open container is creating a key
-    client = OzoneClientFactory.getRpcClient(conf);
-    objectStore = client.getObjectStore();
-    keyString = UUID.randomUUID().toString();
-    volumeName = "testblockdatastreamoutput";
-    bucketName = volumeName;
-    objectStore.createVolume(volumeName);
-    objectStore.getVolume(volumeName).createBucket(bucketName);
+
+    try (OzoneClient client = cluster.newClient()) {
+      ObjectStore objectStore = client.getObjectStore();
+      objectStore.createVolume(VOLUME_NAME);
+      objectStore.getVolume(VOLUME_NAME).createBucket(BUCKET_NAME);
+    }
+
+    return cluster;
   }
 
-  static String getKeyName() {
-    return UUID.randomUUID().toString();
+  private static Stream<Arguments> clientParameters() {
+    return Stream.of(
+        Arguments.of(true),
+        Arguments.of(false)
+    );
   }
 
-  @AfterAll
-  public static void shutdown() {
-    IOUtils.closeQuietly(client);
-    if (cluster != null) {
-      cluster.shutdown();
-    }
+  private static Stream<Arguments> dataLengthParameters() {
+    return Stream.of(
+        Arguments.of(CHUNK_SIZE / 2),
+        Arguments.of(CHUNK_SIZE),
+        Arguments.of(CHUNK_SIZE + 50),
+        Arguments.of(BLOCK_SIZE + 50)
+    );
   }
 
-  @Test
-  public void testHalfChunkWrite() throws Exception {
-    testWrite(chunkSize / 2);
-    testWriteWithFailure(chunkSize / 2);
+  static OzoneClientConfig newClientConfig(ConfigurationSource source,
+                                           boolean flushDelay) {
+    OzoneClientConfig clientConfig = source.getObject(OzoneClientConfig.class);
+    clientConfig.setChecksumType(ContainerProtos.ChecksumType.NONE);
+    clientConfig.setStreamBufferFlushDelay(flushDelay);
+    return clientConfig;
   }
 
-  @Test
-  public void testSingleChunkWrite() throws Exception {
-    testWrite(chunkSize);
-    testWriteWithFailure(chunkSize);
+  static OzoneClient newClient(OzoneConfiguration conf,
+                               OzoneClientConfig config) throws IOException {
+    OzoneConfiguration copy = new OzoneConfiguration(conf);
+    copy.setFromObject(config);
+    return OzoneClientFactory.getRpcClient(copy);
   }
 
-  @Test
-  public void testMultiChunkWrite() throws Exception {
-    testWrite(chunkSize + 50);
-    testWriteWithFailure(chunkSize + 50);
+  @BeforeAll
+  public void init() throws Exception {
+    cluster = createCluster();
+  }
+
+  static String getKeyName() {
+    return UUID.randomUUID().toString();
   }
 
-  @Test
-  @Flaky("HDDS-12027")
```

Comment:
@adoroszlai Do you have any idea on how to keep this while @Flaky tag while it's parameterized?

Reply:
`@Flaky` simply allows the test to be re-run few times if it fails.

1. Strict: only allow that single test case to be repeated by:
   - remove the parameter value that corresponds to old flaky test case
   - restore the old test case
   - replace its implementation to invoke the parameterized method with that value
2. Lenient: mark the parameterized method as `@Flaky`, test cases that always pass won't be re-run.

Reply:
Yes.

Reply:
@chiacyu Please help to add back the `@Flaky` annotation to `testStreamWrite`.